### DISCUSSION TITLE
Problems at the WRO german finals

### DISCUSSION THREAD

**Discussion by Kermit647** [pybricks/support#1680] [General]
Hi!

I have been at the German World Robot Olympiad finals for the second time with Pybricks. What I have seen is that the usage of Pybricks has greatly increased compared to last year, and there have been fewer EV3 and more Spike teams.

With this development came a big problem: While some teams had no issues at all, some teams experienced massive problems with the Bluetooth connection. Some kids were unable to connect, some had problems with the download, and some even had their Bluetooth connection hijacked by other teams (intentional or not). These kids, as young as 8 years old, are in a high-stress situation and are totally unable to debug the Bluetooth problems, and the coaches (parents) are not allowed to interact in any way.

Conclusion : 
We really, really need USB download on the Spike !!!

**Comment by laurensvalk** [comment]
Thanks for your message. It's definitely been noted and we are looking into it :)

**Comment by FLL-Team-24277** [comment]
Alternatively, could there be a key exchange or something where a computer has to use USB to establish a secure pairing, but once that is done, then BT pairing is authorized?

**Reply by laurensvalk** [reply]
We recently added the ability to toggle Bluetooth off/on, and have your preference persist even after reboot.

While not exactly what you asked, I suppose this could be a big help in preventing unwanted connections too. If you want to try it out before we release a new beta, just follow [these instructions](https://pybricks.com/install/spike-mindstorms/#installing-the-latest-build-advanced) to install the preview version. Everything works as usual, but you can press the Bluetooth button to toggle Bluetooth off or on.

Would love to hear your feedback if you try this out.

cc @Kermit647

**Reply by Kermit647** [reply]
Thanks, thats a good thing and will make hijacking the connection much harder, but will not help in a rf noisy enviroment to get a stable connection.

**Comment by djklesk** [comment]
Im a robotics teacher in germany, we had the same problem and worst part it was that the code from the notebook was deleted because some smart technican did reset the notebooks! can u plays add another button in the pybricks code to download code from the hub to the pc?

**Reply by laurensvalk** [reply]
The code is saved on the hub in a compiled/minimized format for efficiency, which means you can't get the original code back to the PC, I'm afraid.
